### PR TITLE
Tests: Stage User Tracker implementation

### PR DESCRIPTION
**Pull request by gkaihorodova**
Fix provide possibility of creation stage user with minimal values,
with uid not specified. Implementation is the same as for User Tracker.

https://fedorahosted.org/freeipa/ticket/6448

### CODE REVIEW

**Comment by mirielka** [comment]
Review notes: same as in https://github.com/freeipa/freeipa/pull/181

**Comment by martbab** [comment]
Bump for review.

**Comment by martbab** [comment]
Can we get this PR rebased against current master to see whether Travis CI is okay. The current checks are polluted by regression that should be fixed.

File: ipatests/test_xmlrpc/test_stageuser_plugin.py [hunk 1]
```diff
@@ -191,6 +196,12 @@ def test_activate_nonexistent(self, stageduser):
 
 @pytest.mark.tier1
 class TestStagedUser(XMLRPC_test):
+    def test_create_with_min_values(self, stageduser_min):
+            """ Create user with uid not specified """
+            stageduser_min.ensure_missing()
+            command = stageduser_min.make_create_command()
+            command()
+
```
Comment (stlaz) [inline ipatests/test_xmlrpc/test_stageuser_plugin.py]: Please fix the indentation here.

File: ipatests/test_xmlrpc/tracker/stageuser_plugin.py [hunk 2]
```diff
@@ -61,23 +61,43 @@ class StageUserTracker(Tracker):
...
 
-    def __init__(self, name, givenname, sn, **kwargs):
+    def __init__(self, name=None, givenname=None, sn=None, **kwargs):
+        """ Check for non-empty unicode string for the required attributes
+        in the init method """
+
+        if not (isinstance(givenname, six.string_types) and givenname):
+            raise ValueError("Invalid first name provided: %r" % givenname)
+        if not (isinstance(sn, six.string_types) and sn):
+            raise ValueError("Invalid second name provided: %r" % sn)
```
Comment (stlaz) [inline ipatests/test_xmlrpc/tracker/stageuser_plugin.py]: Please use str.format() instead in new code.

**Comment by stlaz** [comment]
I just wanted to remove the ACK till @martbab's comment is worked in so nobody pushes it but I found some minor issues that I would like to see fixed in the rebase as well.

File: ipatests/test_xmlrpc/tracker/stageuser_plugin.py (hunk 2 above)
Comment (gkaihorodova) [inline ipatests/test_xmlrpc/tracker/stageuser_plugin.py]: ok, will apply same for #181 as well

**Comment by mbasti-rh** [comment]
Needs rebase

```
Applying: Unaccessible variable self.attrs in Tracker
Patch failed at 0001 Unaccessible variable self.attrs in Tracker
The copy of the patch that failed is found in: .git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
error: patch failed: ipatests/test_xmlrpc/tracker/base.py:76
error: ipatests/test_xmlrpc/tracker/base.py: patch does not apply
```

**Comment by martbab** [comment]
@gkaihorodova the PR cannot be pushed in current form because the first commit 298e1a136c6a430e8deaa558a946ba51874ffd95 is already pushed to master.

So to rebase it correctly please do the following:

Pull the changes from the remote repo (or any other label you have for it) into your local master branch:

```shell
$ git  checkout master; git pull
```

Then do the rebase against the refreshed master branch. The first commit should now disappear as git should detect that it is already there. If not, then abort the current rebase, re-start it in interactive mode (git rebase -i master) and remove the first commit manually (just remove the first line). Then force-push the changes into your fork:

```shell
$ git push -f origin fix-for-6448
```

**Comment by MartinBasti** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/a336de630e9d1ef95a507cc3ee9200c001ab9193
https://fedorahosted.org/freeipa/changeset/c391f6ba58a61e046e49e1b4526b62d7ce250301

**Pull request closed without merging**